### PR TITLE
Fix tab closing behaviors

### PR DESCRIPTION
**Pull request by barijaona**
Fix issues #1667 and #1664

### CODE REVIEW

**Comment by TAKeanice** [comment]
@barijaona I corrected the retain cycle from the link hover displaying functionality, and removed what was not necessary in the deallocation. Since I rearranged the commits, you need to reset your branch to the current remote branch. Before I did this I verified that the issue with retained tabs existed before your commits, and that it did not exist any more after my changes to your branch.

**Comment by barijaona** [comment]
I'm OK with @TAKeanice's changes.
However, as I am the initial author of the pull request, Github does not allow me to approve.

**Review by TAKeanice — APPROVED** [review]

**Comment by TAKeanice** [comment]
> However, as I am the initial author of the pull request, Github does not allow me to approve. 

Strangely, I can as a contributor in this PR.

File: Vienna/Sources/Main window/CustomWKHoverUIDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+//TODO: to me it looks like this protocol should be part of the CustomWKUIDelegate, but the unified display view does not declare conformance to that protocol. Probably that is a design flaw in how the unified display view integrates the browser.
```
[the remark this answers is not in the document]
Comment (TAKeanice) [reply Vienna/Sources/Main window/CustomWKHoverUIDelegate.swift:22]: Sure, I left it in there as an architectural note for later. You can remove it if you prefer not having unfocused todos in the code.

**Pull request merged**